sdk/core/AzureCore/Source/Util/ReachabilityManager.swift

```diff
@@ -101,13 +101,10 @@ import Foundation
 
         public func registerListener(_ listener: @escaping ReachabilityStatusListener) {
             self.listener = { status in
-                // no nothing on the initial status update to force the developer to resume transfers
-                // TODO: Maybe this should be set in a more thoughtful manner
-                guard self.initialized else {
-                    self.initialized = true
-                    return
-                }
+                guard self.previousStatus != .unknown,
```

Comment:
If `previousStatus` is unknown, we still need to update it. Otherwise it will never move out of the `unknown` state.